code/datums/components/hide_weather_planes.dm

```diff
@@ -0,0 +1,116 @@
+/// Component that manages a list of plane masters that are dependent on weather
+/// Force hides/shows them depending on the weather activity of their z stack
+/// Applied to the plane master group that owns them
+/datum/component/hide_weather_planes
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/list/datum/weather/active_weather = list()
+	var/list/atom/movable/screen/plane_master/plane_masters = list()
+
+/datum/component/hide_weather_planes/Initialize(atom/movable/screen/plane_master/care_about)
+	if(!istype(parent, /datum/plane_master_group))
+		return COMPONENT_INCOMPATIBLE
+	var/datum/plane_master_group/home = parent
+	plane_masters += care_about
+	RegisterSignal(care_about, COMSIG_QDELETING, PROC_REF(plane_master_deleted))
+
+	var/list/starting_signals = list()
+	var/list/ending_signals = list()
+	for(var/datum/weather/weather_type as anything in typesof(/datum/weather))
+		starting_signals += COMSIG_WEATHER_TELEGRAPH(weather_type)
+		ending_signals += COMSIG_WEATHER_END(weather_type)
+
+	RegisterSignals(SSdcs, starting_signals, PROC_REF(weather_started))
+	RegisterSignals(SSdcs, ending_signals, PROC_REF(weather_finished))
+
+	if(home.our_hud)
+		attach_hud(home.our_hud)
+	else
+		RegisterSignal(home, COMSIG_GROUP_HUD_CHANGED, PROC_REF(new_hud_attached))
+
+/datum/component/hide_weather_planes/Destroy(force)
+	hide_planes()
+	active_weather = null
+	plane_masters = null
+	return ..()
+
+/datum/component/hide_weather_planes/InheritComponent(datum/component/new_comp, i_am_original, atom/movable/screen/plane_master/care_about)
+	if(!i_am_original)
+		return
+	plane_masters += care_about
+	RegisterSignal(care_about, COMSIG_QDELETING, PROC_REF(plane_master_deleted))
+	if(length(active_weather))
+		care_about.enable_alpha()
+	else
+		care_about.disable_alpha()
```

Comment:
there should be no reason to have planes in our plane masters list if they aren't being used, alphaing them out still incurs the cost of rendering them on the client. That's an important job this component should also be doing

Reply:
the primary problem I was facing with hiding/unhiding them was that they would not update their appearance when on a higher z-level which i think had to do with the out_of_bounds calculating? My understanding was that alpha was working because it would continue to draw and render updates as you transition z-levels

Reply:
did we get this resolved via dms I cannot recall